system-addon/content-src/components/TopSites/_TopSites.scss

```diff
@@ -218,6 +218,54 @@
       @include context-menu-button-hover;
     }
   }
+
+  &.top-sites-1rows {
```

Comment:
I don't really like the approach of needing to declare in css ahead of time each of the `#rows` as it wouldn't allow for about:config custom rows... but maybe oh well. We probably don't want to do `max-height` stuff as `overflow: hidden` has odd cropping at edges.

Alternatively is to have javascript calculate and skip rendering some on resize. What to do for prerendered... mmmm... Maybe with react 16, we prerender with 8 and on mount, re-render without the last 2? Although that will probably have some jank with a flash of a partial row for narrow windows…

@k88hudson any suggestions?

Reply:
Somebody setting more than 4 rows *probably* has a wider layout? And they can probably figure out how to inject their own CSS 😉